### ISSUE TITLE
Question: Which Policy to use?

### ISSUE THREAD

**Issue by iyush1993** [varvet/pundit#575]
My SessionsController has edit action as:
```ruby
def edit
    user = User.find(current_user.id)
    authorize user
end
````
I want to authorize it with pundit. So I created a policy session_policy.rb as:
```ruby
class SessionPolicy < ApplicationPolicy
    def edit?
        true if user.id == record.id
    end
end
```

I also have another Policy called `UserPolicy` for `UsersController` with `edit?` policy.

My question is **How does Pundit determine which policy to use?** Does it determine with respect to the controller or the model I pass to authorize? I am asking because my SessionsController edit action is using the `edit?` policy from `UserPolicy`. I thought the controller name has something to do with choosing the policy. 

Newbie to Pundit.
Thank you

**Comment by Linuus** [comment]
Hi!

First off, Pundit determines the policy class to use based on the object you pass to `authorize` _not_ the controller name. You _can_ override this by passing an extra argument, like this: `authorize @user, policy_class: AnotherPolicy`. However, I don't recommend doing this unless it's _really_ necessary. In my opinion this is a sign that you are authorizing the wrong thing.

Moving on.

What is the `SessionsController#create` method? It sounds like it should edit a `Session` (whatever that means)? If it's an action for editing a `User` (which is what you are authorizing) why not use the `UsersController#edit` action?

**Comment by iyush1993** [comment]
Thank you for the response @Linuus 
Right after posting this I found out that I can explicitly specify the Policy that I want to use.

I have used SessionsController#create for logging in and SessionsController#edit for changing the password.

UsersController#edit takes care of other things. But I see your point. And makes sense as well. Yes, SessionsController#edit does edit the user. Thank you for answering my question and for the suggestion.
